components/brave_wallet_ui/common/hooks/use-pending-transaction.ts

```diff
@@ -455,23 +457,39 @@ export const usePendingTransactions = () => {
     }
 
     try {
-      await dispatch(
-        walletApi.endpoints.approveTransaction.initiate({
+      const result = await approveTransaction({
           chainId: transactionInfo.chainId,
           id: transactionInfo.id,
           coinType: getCoinFromTxDataUnion(transactionInfo.txDataUnion),
-          txType: transactionInfo.txType
-        })
-      ).unwrap()
-      dispatch(PanelActions.setSelectedTransactionId(transactionInfo.id))
-      dispatch(PanelActions.navigateTo('transactionStatus'))
+          txType: transactionInfo.txType}).unwrap()
+      if (!result.success) {
+        dispatch(
+          UIActions.setTransactionProviderError({
+            providerError: {
+              code: result.errorUnion,
+              message: result.errorMessage
+            },
+            transactionId: transactionInfo.id
+          })
+        )
+      }
     } catch (error) {
       dispatch(
         UIActions.setTransactionProviderError({
-          providerError: error.toString(),
+          providerError: {
+            code: {
+              ...
+              emptyProviderErrorCodeUnion,
```

Comment:
nit: spread operator should be on the same line as the object that is getting spreaded
```suggestion
              ...emptyProviderErrorCodeUnion,
```